docs/adr/0012-versioning.md

```diff
@@ -0,0 +1,46 @@
+# 12) Versioning {#adr_0012}
+
+Date: 2021-02-08
+
+## Status
+
+accepted
+
+## Context
+
+WMF does during their release process create a number of branches which are used during the release process to orchestrate the different tools to produce the correct artifacts and to support the continuous deployment to production. The release branches (ex. REL1_35) are cut for MediaWiki and every extension repository with the purpose of maintaining releases and the weekly deployment branches are (ex. wmf/1.36.0-wmf.13) are used to deployments to the different [groups](https://versions.toolforge.org/). 
+
+Both of these types of branches are maintained by the WMF and will be used by the wikibase release strategy for our major releases and deployment. However we need to decide how to maintain our own intermediate releases that will be published on releases.wikimedia.org, dockerhub and tagged on git.
+
+As there we will still rely on the release branches and deployment branches, we don't see any real benefit by maintaining an additional set of branches for minor releases. These releases could still be maintained through git tagging the WMDE controlled repositories at the point in time a release is produced ([similar to how mediawiki is maintaining their release candidates](https://gerrit.wikimedia.org/g/mediawiki/core/+/refs/tags/1.35.0-rc.0)). 
+
+
+As for the tag itself we also need to decide on version-numbering scheme for differentiating releases where the WMF version number does not change. Seeing that MediaWiki is already using the semantic versioning for incrementing their (maintenance releases) releases in between new release branches being cut it could make sense for wikibase to adopt a similar process as we will still rely on the release branches for future releases.
```

Comment:
> that MediaWiki is already using the semantic versioning

I don't think it is accurate. Mediawiki does not use semantic versioning I believe. They do use three-part version numbers, but this does not imply following semantic versioning. There are breaking changes made etc, and the "major" version number has ever changed.

https://www.mediawiki.org/wiki/Version_lifecycle talks about "major releases" to refer to versions 1.35, 1.36, etc. In semver's language this  does not match, as 35, 36 would be _minor_ versions.
I didn't find MW's/WMF's official name for the last  version number part.

I am intending to make a constructive suggestion but I realize now I don't full understand the point of this second sentence here.
Is it saying that as Mediawiki already uses a version numbering scheme which increments the "small" version number when there are updates (e.g. security fixes) added to the "stable release" (e.g. 1.35), and that there are git branches made using the similar naming pattern? And that could use something similar?

Reply:
I think this whole section was my first proposal that got shot down by @tarrow. Essentially what I suggested was lets just keep using the versioning schema already in place (semantic or not) and call our next release 1.35.0 the next one in march would then be 1.35.1